File: backend/src/main/java/com/back/fairytale/domain/keyword/repository/KeywordRepository.java
package com.back.fairytale.domain.keyword.repository;

import com.back.fairytale.domain.keyword.entity.Keyword;
import com.back.fairytale.domain.keyword.enums.KeywordType;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.List;
import java.util.Optional;

public interface KeywordRepository extends JpaRepository<Keyword, Long> {
    Optional<Keyword> findByKeyword(String keyword);
    List<Keyword> findByKeywordType(KeywordType keywordType);
    Optional<Keyword> findByKeywordAndKeywordType(String keyword, KeywordType keywordType);
}

File: backend/src/main/java/com/back/fairytale/domain/keyword/repository/KeywordRepository.kt
package com.back.fairytale.domain.keyword.repository

import com.back.fairytale.domain.keyword.entity.Keyword
import com.back.fairytale.domain.keyword.enums.KeywordType
import org.springframework.data.jpa.repository.JpaRepository
import java.util.Optional

interface KeywordRepository : JpaRepository<Keyword, Long> {
    fun findByKeyword(keyword: String): Optional<Keyword>
    fun findByKeywordType(keywordType: KeywordType): List<Keyword>
    fun findByKeywordAndKeywordType(keyword: String, keywordType: KeywordType): Optional<Keyword>
}
